File: java/src/main/java/pro/wata/sdk/digitalgoods/deposit/DepositClient.java
package pro.wata.sdk.digitalgoods.deposit;

import pro.wata.sdk.http.HttpTransport;

/**
 * Депозитные операции, общие для Steam/Top-Up/ваучеров: баланс депозита и
 * статус депозитного заказа. Каждый продуктовый клиент ({@code SteamClient},
 * {@code TopupClient}, {@code VouchersClient}) держит свой экземпляр,
 * построенный на транспорте со своим токеном — токены между продуктами не
 * переиспользуются, в том числе здесь.
 */
public final class DepositClient {

    private final HttpTransport transport;

    public DepositClient(HttpTransport transport) {
        this.transport = transport;
    }

    /** {@code GET /v1/deposit/balance}. */
    public DepositBalance balance() {
        return transport.get("/v1/deposit/balance", null, DepositBalance.class);
    }

    /** {@code GET /v1/deposit/order/{orderId}}. */
    public DepositOrder getOrder(String orderId) {
        return transport.get("/v1/deposit/order/" + orderId, null, DepositOrder.class);
    }
}

File: dotnet/src/Wata.Sdk/DigitalGoods/Deposit/DepositClient.cs
using Wata.Sdk.Errors;
using Wata.Sdk.Http;

namespace Wata.Sdk.DigitalGoods.Deposit;

/// <summary>
/// Общая инфраструктура депозитной оплаты (баланс мерчанта, статус депозитного заказа).
/// Эти эндпоинты общие для Steam/Top-Up/ваучеров, оплачиваемых с депозита, поэтому
/// клиент использует первый сконфигурированный токен из SteamToken, TopUpToken,
/// VouchersToken — какой у вас есть, тот и подойдёт для проверки баланса и статуса.
/// Если ни один из этих токенов не задан, обращение бросает WataConfigurationException.
/// </summary>
public sealed class DepositClient
{
    private readonly WataHttpTransport? _transport;

    internal DepositClient(WataHttpTransport? transport) => _transport = transport;

    private WataHttpTransport Transport => _transport ?? throw new WataConfigurationException(
        "Для работы с депозитом цифровых товаров не задан ни один из токенов: SteamToken, TopUpToken, VouchersToken. " +
        "Передайте хотя бы один из них при создании WataClient.");

    /// <summary>GET /v1/deposit/balance — баланс депозита мерчанта.</summary>
    public Task<DepositBalance> GetBalanceAsync(CancellationToken cancellationToken = default) =>
        Transport.SendAsync<DepositBalance>(HttpMethod.Get, "/api/v1/deposit/balance", null, null, requiresAuth: true, allowRetry: true, cancellationToken);

    /// <summary>GET /v1/deposit/order/{orderId} — общий статус депозитного заказа (Steam/Top-Up/ваучеры).</summary>
    public Task<DepositOrder> GetOrderAsync(string orderId, CancellationToken cancellationToken = default) =>
        Transport.SendAsync<DepositOrder>(HttpMethod.Get, $"/api/v1/deposit/order/{Uri.EscapeDataString(orderId)}", null, null, requiresAuth: true, allowRetry: true, cancellationToken);
}
